includes/filerepo/file/LocalFile.php

```diff
@@ -160,6 +167,8 @@ function __construct( $title, $repo ) {
 	/**
 	 * Get the memcached key for the main data for this file, or false if
 	 * there is no access to the shared cache.
+	 *
+	 * @return string|bool
```

Comment:
Not really a comment, just a sigh to be taken into consideration. Typecasting and functions returning values of multiple types. I agree, that cache keys might not be the case. But in general in such caches we would have to check for type, as strings `''` and `'0'` evaluate to boolean `false`. I am more into _return one specific type or null_ and `!is_null()` checks convention. I treat boolean values as answers to questions - that should never be `null`.